### ISSUE TITLE
For loop keywords used outside for loop

### ISSUE THREAD

**Issue by bhirsz** [MarketSquare/robotframework-robocop#477]
Keywords like Continue For Loop should not be used outside for loop.

**Comment by bhirsz** [comment]
How to implement it:
    - when inside For loop, set flag (ie `in_loop`) to True, set to False after vising it:
      ```
      def visit_For(self, node):
           self.in_loop = True
           self.generic_visit(self)  # visit childs 
           self.in_loop = False
      ```
Then it's possible to just visit KeywordCalls, normalize the name and if it's for loop keyword and we're not in loop, report it:
```
    def visit_KeywordCall(self, node):
        if self.in_loop and self.normalize_name(node.name) in self.loop_keywords:
            self.report( ... )
```

`normalize_name` should lowercase and replace spaces and underscores with ''
`loop_keywords` should be set of for loop keywords such as 'continueforloop', 'continueforloopif', 'exitforloop', 'exitforloopif'.

**Comment by bhirsz** [comment]
Related, it will probably print parsing error
https://github.com/robotframework/robotframework/issues/4185

**Comment by bhirsz** [comment]
Starting work. I will also include BREAK/CONTINUE in it -> currently there are raised as parsing error but it should be filtered out and reported by the same rule for Exit For Loop/Continue For Loop variants.
